packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/classes.ts

```diff
@@ -206,10 +220,17 @@ export abstract class HandlerFrameworkClass extends ClassType {
           ['handler', expr.lit(props.handler)],
           ['runtime', this.buildRuntimeProperty(scope, { runtime: props.runtime, isEvalNodejsProvider })],
         ]);
+        const metadataStatements: Statement[] = [
+          expr.directCode(`this.addMetadata('${CUSTOM_RESOURCE_SINGLETON}', true)`),
+          expr.directCode(`if (props?.logGroup) { this.logGroup.node.addMetadata('${CUSTOM_RESOURCE_SINGLETON_LOG_GROUP}', true) }`),
+          // (this singleton function) as it extends FunctionBase and property lambdaFunction is available to be cast to lambda.Function to access the internal readonly _logRetention
```

Comment:
```suggestion
          // We need to access the private `_logRetention` custom resource, the only public property - `logGroup` - provides an ARN reference to the resource, instead of the resource itself. 
```